src/com/xilinx/rapidwright/rwroute/GlobalSignalRouting.java

```diff
@@ -386,6 +504,17 @@ public static Map<RouteNode, List<SitePinInst>> getLCBPinMappings(List<SitePinIn
         return lcbMappings;
     }
 
+    private static boolean centroidEligible(ClockRegion candidate) {
+        // Check if we can use this clock region
+        Tile lowerLeft = candidate.getLowerLeft();
+        Tile upperLeft = candidate.getUpperLeft();
+        int tileHeight = 0;
+        if (lowerLeft != null && upperLeft != null) {
+            tileHeight = Math.abs(lowerLeft.getRow() - upperLeft.getRow());
+        }
+        return tileHeight > 40;
```

Comment:
Where does 40 come from?

Reply:
This is the minimum height of a clock region that we would want to attempt using as a centroid candidate.  The problem is that there are a lot of CRs in Versal devices that have very small CRs that don't have the clocking resources available.

Reply:
I've updated the code to define 40 as a named static variable.